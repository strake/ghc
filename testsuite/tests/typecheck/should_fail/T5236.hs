{-# LANGUAGE MultiParamTypeClasses, FunctionalDependencies, FlexibleContexts  #-}

module T5236 where

data A
data B

class Id a b | a -> b, b -> a

instance Id A A
instance Id B B

loop :: Id A B => Bool
loop = True

-- f :: Bool
-- f = loop
