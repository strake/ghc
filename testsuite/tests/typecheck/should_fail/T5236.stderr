
T5236.hs:13:9:
    Couldn't match type ‘B’ with ‘A’
    arising from a functional dependency between:
      constraint ‘Id A B’
        arising from the type signature for loop :: Id A B => Bool
      instance ‘Id B B’ at T5236.hs:11:10-15
    In the ambiguity check for: Id A B => Bool
    To defer the ambiguity check to use sites, enable AllowAmbiguousTypes
    In the type signature for ‘loop’: loop :: Id A B => Bool
